waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -77,21 +78,25 @@ analyzeWaspProject ::
   CompileOptions ->
   IO (Either [CompileError] AS.AppSpec, [CompileWarning])
 analyzeWaspProject waspDir options = do
-  waspFilePathOrError <- left (: []) <$> findWaspFile waspDir
+  dirResult <- waspDirExists waspDir
+  case dirResult of
+    Left err -> return (Left [err], [])
+    Right _ -> do
+      waspFilePathOrError <- left (: []) <$> findWaspFile waspDir
```

Comment:
A better place to handle this check is the function `findWaspFile`.

After all, we don't care about the directory at all. All we care about is:
- Do the users have a `*.wasp` file?
- If they do, is it called something other than just `.wasp`?
- If not, tell them to name it properly.

That's on me, the issue could have been explained better. Sorry for the troubles :)

So, to summarize one last time:
> While looking for a `*.wasp` file, we should also ensure that the found file isn't called `.wasp`

While you're at it, you might also want to check whether there are multiple `.wasp` files and report that. 

Because, in theory, Wasp will only find one of  `*.wasp` files. Let's say we have two of them: `something.wasp` and `.wasp`. `findWaspFile` could return the `something.wasp`, letting `.wasp` slip under the radar and cause problems later.

And even if it didn't, allowing multiple `.wasp` files can be confusing.